processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -337,6 +336,41 @@ public String apply(@Nullable String input)
     return makeIndexFiles(indexes, outDir, progress, mergedDimensions, mergedMetrics, rowMergerFn, indexSpec);
   }
 
+  // I had seen some strange problem in some segments where this version didn't seem to pick up the new compression properly.
+  // I have not been able to reproduce it, but have not seen it in the IndexMaker version, so until this is tested more
+  // thoroughly (like with old segments or v9 segments with old null handlings) the generic "safe" convert should be
+  // IndexMaker's version
+  public static File convert(final File inDir, final File outDir, final IndexSpec indexSpec) throws IOException
```

Comment:
The unit tests show they work fine for "modern" segments. I just have absolutely no way of testing against old format segments.